poetry/utils/shell.py

```diff
@@ -41,7 +42,16 @@ def get(cls):  # type: () -> Shell
 
         try:
             name, path = detect_shell(os.getpid())
-        except (RuntimeError, ShellDetectionFailure):
+        except ShellDetectionFailure:
+            if os.name == "posix":
+                shell = os.environ["SHELL"]
+            elif os.name == "nt":
+                shell = os.environ["COMSPEC"]
+            else:
+                raise RuntimeError
```

Comment:
Should this maybe be a custom `UnknownShellError` or something like it?

Reply:
I think it's more simply to just repeat the error message here. A custom `ErrorClass` looks a bit to much to me here.

Thanks a lot for your valuable input :+1: